src/main/java/me/ryanhamshire/GriefPrevention/BlockEventHandler.java

```diff
@@ -232,6 +233,22 @@ public void onBlockPlace(BlockPlaceEvent placeEvent)
 		String noBuildReason = GriefPrevention.instance.allowBuild(player, block.getLocation(), block.getType());
 		if(noBuildReason != null)
 		{
+			// Allow players with container trust to place books in lecterns
+			PlayerData playerData = this.dataStore.getPlayerData(player.getUniqueId());
+			Claim claim = this.dataStore.getClaimAt(block.getLocation(), true, playerData.lastClaim);
+			if (block.getType() == Material.LECTERN && placeEvent.getBlockReplacedState() instanceof Lectern)
+			{
+				if (claim.allowContainers(player) != null)
+				{
+					placeEvent.setCancelled(true);
+					GriefPrevention.sendMessage(player, TextMode.Err, claim.allowContainers(player));
```

Comment:
Perhaps minor, but you could store the result of `claim.allowContainers(player)` above and reuse the result, rather than running that check twice.

Reply:
You're right definitely better.